src/test/java/org/commcare/formplayer/tests/BaseTestClass.java

```diff
@@ -801,6 +801,21 @@ <T> T sessionNavigate(String[] selections, String testName, String locale, Class
                 clazz);
     }
 
+    SyncDbResponseBean intervalSyncDB(String testName, String restoreAs) throws Exception {
```

Comment:
This is using an old style of testing. There is a section in the DEV_README about how testing should be done: https://github.com/dimagi/formplayer/blob/master/DEV_GUIDE.md#testing

Here's an example test in the new style: https://github.com/dimagi/formplayer/blob/0a4e9aa9db3d7943752a64c5845f2f3a837e7513/src/test/java/org/commcare/formplayer/tests/MenuAssertionsTest.java#L39

You'll need to modify the [SyncDbRequest](https://github.com/dimagi/formplayer/blob/a42e5105de1457cbc8e6d026d0d6c8d606fddedc/src/test/java/org/commcare/formplayer/junit/request/SyncDbRequest.kt#L12) mock or create a new mock for this endpoint (I think only the URL will be different).

Reply:
I've been trying to migrate `testSyncDb` and finally was able to get the test to run but I'm getting test failure on `assert (15 == caseStorage.getNumRecords());`

Do you have an idea why it's failing? These are the change I made https://github.com/dimagi/formplayer/compare/jt/interval-sync...jt/interval-sync-test?expand=1